ERCS/erc-7087.md

```diff
@@ -0,0 +1,163 @@
+- `mime.dataurl`, which indicates to decode the returned bytes as a [RFC 2397](https://www.rfc-editor.org/rfc/rfc2397) data URL. After decoding, the decoded body will be returned as the main output, and the MIME type specified in the data URL will be used. If the data cannot be parsed as data URL, an error will be returned.
+  
+If multiple query parameters are present, the last query parameter will be applied.  If none of the query parameter is specified, `Content-Type` is defined by [ERC-6860](./eip-6860.md).  If the `returns` query parameter is specified, the `mime.xxx` parameters will be ignored and the `Content-Type` will be defined by [ERC-6860](./eip-6860.md).
+
+In [RFC 2234](https://www.rfc-editor.org/rfc/rfc2234) ABNF notation, the [ERC-6860](./eip-6860.md) syntax is :
+
+```
+attribute       = attrName "=" attrValue
+attrName        = "returns"
+                / "returnTypes"
+attrValue       = [ "(" [ retTypes ] ")" ]
+```
+
+This standard evolves it into : 
+
+```
+attribute       = retAttr / mimeCAttr / mimeTAttr / mimeDAttr
+retAttr         = retAttrName "=" retAttrValue
+retAttrName     = "returns"
+                / "returnTypes"
+retAttrValue    = [ "(" [ retTypes ] ")" ]
+
+mimeCAttr       = "mime.content=" mimeCAttrVal
+mimeCAttrVal    = # ABNF of MIME type as in RFC 6838 
+mimeTAttr       = "mime.type=" 1*( ALPHA / DIGIT )
+mimeDAttr       = "mime.dataurl"
+```
+
+
+### ``?returns=`` field naming
+
+[ERC-6860](./eip-6860.md) defines the syntax of ``?returns=``. This standard evolves this syntax.
+
+In [RFC 2234](https://www.rfc-editor.org/rfc/rfc2234) ABNF notation, the [ERC-6860](./eip-6860.md) syntax is :
+
+```
+retType         = retRawType *( "[" [ %x31-39 *DIGIT ] "]" )
+```
+
+This standard evolves it into : 
+
+```
+retType         = [ retFieldName ":" ] retRawType *( "[" [ %x31-39 *DIGIT ] "]" )
+retFieldName    = 1*retFieldChar
+                \ %x22 1*( retFieldChar / retTypeReserved ) %x22 ; %x22 being double quote
+retFieldChar    = unreserved / pct-encoded
+                / "!" / "$" / "'" / "*" / "+" / ";" / "@"
+retTypeReserved = ":" / "," / "(" / ")"
+
+```
+
+When field names are specified, the JSON returned is no longer an array but an object. As an basic example, ``?returns=(field1:uint)`` will return ``{"field1": 1234}``. 
+
+If ``?returns=`` specifies multiple types, all must have a field name. At a tuple definition level or at the base level, if one type does not have a field name, then the formatting revert to a array at this level. As a basic example, ``?returns=(field1:(subField1:uint,string))`` will return ``{"field1": [1234, "abcd"]}``. 
+
+At a tuple definition level or at the base level, if a field name is duplicate, an error will be throw.
+
```

Comment:
You should define what "an error will be thrown" means. Is that an HTTP error code?

Reply:
To be adressed on the splitted ERC -- but basically no, it would be an error like errors you get when the domain name cannot be resolved, or the certificate is wrong : it doesn't even reach the stage of getting an answer with HTTP code, it stops before.

Reply:
"error is thrown" still exists in the updated proposal, but I don't see this definition in it. I'd recommend explaining it in the body of the document.

Reply:
Sorry, I thought there were all on the other section.
I have replaced "an error is thrown" with "the URL is not fetched and an error message is displayed to the user".